query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryRequestUtil.java

```diff
@@ -102,37 +101,33 @@ public static Expression createStringArrayLiteralValueExpression(List<String> va
         .build();
   }
 
-  public static boolean isComplexAttributeExpression(Expression expression) {
-    return expression.getValueCase() == ValueCase.ATTRIBUTE_EXPRESSION
-        && expression.getAttributeExpression().hasSubpath();
-  }
-
-  private static boolean isColumnMapAttribute(
-      Expression expression, ViewDefinition viewDefinition) {
-    return expression.getValueCase() == COLUMNIDENTIFIER
-        && isMapField(expression.getColumnIdentifier().getColumnName(), viewDefinition);
-  }
-
-  public static boolean isMapSubpathAttributeExpression(
+  public static boolean isMapAttributeExpression(
       Expression expression, ViewDefinition viewDefinition) {
-    return isComplexAttributeExpression(expression)
-        && isMapField(expression.getAttributeExpression().getAttributeId(), viewDefinition);
+    switch (expression.getValueCase()) {
+      case COLUMNIDENTIFIER:
+        return isMapField(expression.getColumnIdentifier().getColumnName(), viewDefinition);
+      case ATTRIBUTE_EXPRESSION:
+        return expression.getAttributeExpression().hasSubpath()
+            && isMapField(expression.getAttributeExpression().getAttributeId(), viewDefinition);
+      default:
+        return false;
+    }
   }
 
-  public static boolean isMapAttributeExpression(
+  public static boolean isSimpleAttributeExpression(
       Expression expression, ViewDefinition viewDefinition) {
-    return isColumnMapAttribute(expression, viewDefinition)
-        || isMapSubpathAttributeExpression(expression, viewDefinition);
+    switch (expression.getValueCase()) {
+      case COLUMNIDENTIFIER:
+        return true;
+      case ATTRIBUTE_EXPRESSION:
+        return !expression.getAttributeExpression().hasSubpath()
+            || !isMapField(expression.getAttributeExpression().getAttributeId(), viewDefinition);
+      default:
+        return false;
+    }
   }
 
   private static boolean isMapField(String columnName, ViewDefinition viewDefinition) {
     return viewDefinition.getColumnType(columnName) == ValueType.STRING_MAP;
   }
-
-  public static boolean isSimpleAttributeExpression(
```

Comment:
This function can be removed too. In place we can use !isMapAttributeExpression() where some tight condition is used.